packages/block-library/src/navigation/edit.js

```diff
@@ -119,132 +84,247 @@ function Navigation( {
+								title: __( 'Justify items center' ),
+								isActive:
+									'center' === attributes.itemsJustification,
+								onClick: handleItemsAlignment( 'center' ),
+							},
+							{
+								icon: navIcons.justifyRightIcon,
+								title: __( 'Justify items right' ),
+								isActive:
+									'right' === attributes.itemsJustification,
+								onClick: handleItemsAlignment( 'right' ),
+							},
+						] }
+					/>
+				</BlockControls>
+			</>
+		);
+	},
+	'withBlockControls'
+);
+
+addFilter(
+	'navigation.BlockEdit',
+	'core/block-library/navigation/with-block-controls',
+	withBlockControls
+);
+
+export const withListView = createHigherOrderComponent(
+	( BlockEdit ) => ( props ) => {
+		const { clientId } = props;
+		const { navigatorToolbarButton, navigatorModal } = useBlockNavigator(
+			clientId
+		);
+
+		return (
+			<>
+				<BlockEdit { ...props } />
+				<BlockControls>
+					<ToolbarGroup>{ navigatorToolbarButton }</ToolbarGroup>
+				</BlockControls>
+				{ navigatorModal }
+			</>
+		);
+	},
+	'withListView'
+);
+
+addFilter(
+	'navigation.BlockEdit',
+	'core/block-library/navigation/with-list-view',
+	withListView
+);
+
+export const withFormattingControls = createHigherOrderComponent(
+	( BlockEdit ) => ( props ) => {
 		const {
-			getClientIdsOfDescendants,
-			hasSelectedInnerBlock,
-			getSelectedBlockClientId,
-		} = select( 'core/block-editor' );
-		const isImmediateParentOfSelectedBlock = hasSelectedInnerBlock(
-			clientId,
-			false
+			TextColor,
+			BackgroundColor,
+			ColorPanel,
+		} = __experimentalUseColors(
+			[
+				{ name: 'textColor', property: 'color' },
+				{ name: 'backgroundColor', className: 'has-background' },
+			],
+			{
+				contrastCheckers: [
+					{
+						backgroundColor: true,
+						textColor: true,
+					},
+				],
+				colorDetector: { targetRef: ref },
+				colorPanelProps: {
+					initialOpen: true,
+				},
+			}
 		);
-		const selectedBlockId = getSelectedBlockClientId();
-		const selectedBlockHasDescendants = !! getClientIdsOfDescendants( [
-			selectedBlockId,
-		] )?.length;
-		return {
-			isImmediateParentOfSelectedBlock,
-			selectedBlockHasDescendants,
-			hasExistingNavItems: !! innerBlocks.length,
-		};
-	} ),
-	withDispatch( ( dispatch, { clientId } ) => {
-		return {
-			updateInnerBlocks( blocks ) {
-				if ( blocks?.length === 0 ) {
-					return false;
-				}
-				dispatch( 'core/block-editor' ).replaceInnerBlocks(
-					clientId,
-					blocks
-				);
-			},
-		};
-	} ),
-] )( Navigation );
+		return (
+			<>
+				<TextColor>
+					<BackgroundColor>
+						<BlockEdit { ...props } />
+					</BackgroundColor>
+				</TextColor>
+				<BlockControls>
+					<BlockColorsStyleSelector
+						TextColor={ TextColor }
+						BackgroundColor={ BackgroundColor }
+					>
+						{ ColorPanel }
+					</BlockColorsStyleSelector>
+				</BlockControls>
+			</>
+		);
+	},
+	'withFormattingControls'
+);
+
+addFilter(
+	'navigation.BlockEdit',
```

Comment:
Could these features being implemented using the existing `editor.BlockEdit` filter? It looks like all that's needed to achieve that is to add the following early return to the start of the filter function and it would work:
```
		if ( props.name !== 'core/navigation' ) {
			return <BlockEdit { ...props } />;
		}
```